doc/bum_storm_control/bum_storm_control_hld.md

```diff
@@ -0,0 +1,399 @@
+    kbps     = 1*13 DIGIT   ; CIR value in kilo bits per second
+         
+### 3.2.2 APP_DB
+No tables are introduced in APP_DB
+### 3.2.3 STATE_DB
+No tables are introduced in STATE_DB. 
+
+### 3.2.4 ASIC_DB
+No changes are introduced in ASIC_DB. 
+### 3.2.5 COUNTER_DB
+No changes are introduced in COUNTER_DB. 
+
+
+## 3.3 Switch State Service Design
+### 3.3.1 PolicerOrch changes 
+Policer Orchestration agent is responsible for the following activities:
+   - Subscribes to notifications on CFG_PORT_STORM_CONTROL_TABLE entries in the CONFIG_DB. 
+   - Creates an internal name for policer by encoding the interface name and storm-control type. 
+   - Creates the policer based on the parameters and associates the policer identifier to internal name
+   - Populates the port attribute SAI structures and pushes the entry to ASIC_DB. 
+
+
+## 3.4 SAI
+Storm-control SAI interface APIs are already defined. The table below represents the SAI attributes which shall be used for BUM storm-control.
+
+
+###### Table 2: Storm Control SAI attributes
+| Storm Control component  | SAI attributes                                         |                          
+|--------------------------|--------------------------------------------------------|
+| Meter type               | SAI_POLICER_ATTR_METER_TYPE                            |
+| Meter mode               | SAI_POLICER_ATTR_MODE                                  |
+| CIR (bps)                | SAI_POLICER_ATTR_CIR                                   |
+| Unknown-unicast policer  | SAI_PORT_ATTR_FLOOD_STORM_CONTROL_POLICER_ID           |
+| Broadcast policer        | SAI_PORT_ATTR_BROADCAST_STORM_CONTROL_POLICER_ID       |
+| Unknown-Multicast policer| SAI_PORT_ATTR_MULTICAST_STORM_CONTROL_POLICER_ID       |
+
+
+The BUM storm-control feature can be enabled on a physical port. 
+
+The **create_policer** SAI API is used to create the policer based on the policer parameters listed in Table 2.
+   - Meter Type - Can be packets or bytes. Storm-control uses bytes meter type. 
+   - Meter Mode - Can be st_tcm or tr_tcm or storm. Storm-control uses storm meter mode. 
+   - CIR (bps) - Value of CIR given in bps. 
+
+The **set_port_attribute** SAI API is used to set the policer on an interface. 
+   - Unknown-unicast policer - Enables storm-control for unknown-unicast traffic received on the interface. 
+   - Broadcast policer - Enables storm-control for broadcast traffic received on the interface. 
+   - Unknown-multicast policer - Enables storm-control for unknown-multicast traffic received on the interface. 
+
+## 3.5 CLI
+### 3.5.1 Data Models
+Configuration is supported using Click and Klish CLI commands.
+
+
+### 3.5.2 Configuration Commands
+This section lists down the Click commands.
+BUM storm-control can be configured only on physical interfaces.  
+**switch# config interface storm-control {broadcast | unknown-unicast | unknown-multicast} {add|del} \<interface_name\> {\<kilo_bits_per_second\>}**
+
```

Comment:
Suggest to change the SONiC CLI command as follows,
config interface storm-control { add | del } <interface name> {broadcast | unknown-unicast | unknown-multicast} <kilo_bits_per_second>

Reply:
storm-control policer types are keywords. add/del are operations. Keywords are kept in the beginning of the command. 
user might want to know the type before performing the operation.